src/error.rs

```diff
@@ -15,6 +59,22 @@ pub type Result<T> = ::std::result::Result<T, Error>;
 
 impl Error {
     /// Returns a possible URL related to this error.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # extern crate reqwest;
+    /// # fn main() {
+    /// let response = reqwest::get("");
+    /// match response {
+    ///   Err(e) => match e.url() {
+    ///     None => println!("No Url given"),
+    ///     Some(url) => println!("Bad Url given: {}", url),
```

Comment:
Difficult to translate into an example.  I created a hyper server to give me a redirect loop.  I'll post the client example to show the final stop.  Unfortunately it wont be reproducible without a redirect loop server (of which I could not find an online example of a redirect loop)